server/handler.go

```diff
@@ -772,6 +773,50 @@ func (h *Handler) AddScatterRegionOperator(regionID uint64, group string) error
 	return nil
 }
 
+// AddScatterRegionsOperators add operators to scatter regions and return the processed percentage and error
+func (h *Handler) AddScatterRegionsOperators(startRawKey, endRawKey, group string, retryLimit int64) (int, error) {
+	c, err := h.GetRaftCluster()
+	if err != nil {
+		return 0, err
+	}
+	startKey, _, err := keyutil.ParseRawKey(startRawKey)
+	if err != nil {
+		return 0, err
+	}
+	endKey, _, err := keyutil.ParseRawKey(endRawKey)
+	if err != nil {
+		return 0, err
+	}
+	regions := c.ScanRegions(startKey, endKey, -1)
+	regionMap := make(map[uint64]*core.RegionInfo, len(regions))
+	errList := make([]error, len(regions))
+	unProcessedRegionsCount := 0
+	for _, region := range regions {
+		// If region is Hot, add it into unProcessedRegions
+		if c.IsRegionHot(region) {
+			unProcessedRegionsCount++
+			errList = append(errList, errors.Errorf("region %d is a hot region", region.GetID()))
+			continue
+		}
+		regionMap[region.GetID()] = region
+	}
+	failures := make(map[uint64]error, len(regionMap))
+	// If there existed any region failed to relocated after retry, add it into unProcessedRegions
+	ops := c.GetRegionScatter().ScatterRegions(regionMap, failures, group, 0, retryLimit)
+	for _, failureErr := range failures {
+		unProcessedRegionsCount++
+		errList = append(errList, failureErr)
+	}
+	// If there existed any operator failed to be added into Operator Controller, add its regions into unProcessedRegions
+	for _, op := range ops {
+		if ok := c.GetOperatorController().AddOperator(op); !ok {
+			unProcessedRegionsCount++
+			errList = append(errList, errors.WithStack(ErrAddOperator))
+		}
+	}
+	return 100 - (unProcessedRegionsCount/len(regions))*100, errs.AggregateErrors(errList)
```

Comment:
*100 then / len(regions)

Reply:
aggregateErrors could be very very long if there are too many errors.

Reply:
updated.